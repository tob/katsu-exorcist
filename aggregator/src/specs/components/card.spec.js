import React from 'react';
import Card from '../../components/card.js';
import { shallow } from 'enzyme';

let component;
let card;
describe('render method', () => {

  describe('when undefined card', () => {
    beforeEach(() => {
      card = {foo:'foo'};
      component = shallow(<Card card={card} />)
    });
  
    it('should display a broken card', () => {
      expect(component.html()).toMatchSnapshot();
    })
  });

  describe('when video card', () => {
    beforeEach(() => {
      card = {snippet:{resourceId: 'foo'}}
      component = shallow(<Card card={card} />)
    });
  
    it('should have card__video class', () => {
      expect(component.debug()).toMatch(/card__video/);
    })
  })

  describe('when event card', () => {
    beforeEach(() => {
      card = {start: 'whenever'}
      component = shallow(<Card card={card} />)
    });
  
    it('should have card__event class', () => {
      expect(component.debug()).toMatch(/card__event/);
      expect(component.html()).toMatchSnapshot();
    })
  })

  describe('when article card', () => {
    beforeEach(() => {
      card = {resolved_title: 'foo title'}
      component = shallow(<Card card={card} />)
    });
  
    it('should have card__article class', () => {
      expect(component.debug()).toMatch(/card__article/);
    })
  })

});
