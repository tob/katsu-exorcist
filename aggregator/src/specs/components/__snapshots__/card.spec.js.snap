// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render method when event card should have card__event class 1`] = `"<div class=\\"card card__event card--0\\"><span class=\\"card__date\\">Invalid Date</span><span class=\\"card__title\\"></span><span class=\\"card__date\\">private event</span></div>"`;

exports[`render method when undefined card should display a broken card 1`] = `"<div class=\\"card card__undefined card--4\\">some devil touched this</div>"`;
